src/super_gradients/convert_from_recipe.py

```diff
@@ -0,0 +1,179 @@
+"""
+Entry point for converting recipe file to self-contained train.py file.
+
+General use: python -m super_gradients.convert_from_recipe --config-name="DESIRED_RECIPE".
+For recipe's specific instructions and details refer to the recipe's configuration file in the recipes directory.
+"""
+import argparse
+import collections
+import os.path
+from typing import Tuple, Mapping, Dict
+
+import black
+import hydra
+import pkg_resources
+from hydra.core.global_hydra import GlobalHydra
+from omegaconf import DictConfig, OmegaConf, ListConfig
+
+from super_gradients import Trainer
+from super_gradients.common import MultiGPUMode
+from super_gradients.common.abstractions.abstract_logger import get_logger
+from super_gradients.common.environment.omegaconf_utils import register_hydra_resolvers
+from super_gradients.common.environment.path_utils import normalize_path
+from super_gradients.training.utils import get_param
+
+logger = get_logger(__name__)
+
+
+def recursively_walk_and_extract_hydra_targets(cfg, objects=None, prefix=None) -> Tuple[DictConfig, Dict[str, Mapping]]:
+    if objects is None:
+        objects = collections.OrderedDict()
+    if prefix is None:
+        prefix = ""
+
+    if isinstance(cfg, DictConfig):
+        for key, value in cfg.items():
+            value, objects = recursively_walk_and_extract_hydra_targets(value, objects, prefix=f"{prefix}_{key}")
+            cfg[key] = value
+
+        if "_target_" in cfg:
+            target_class = cfg["_target_"]
+            target_params = dict([(k, v) for k, v in cfg.items() if k != "_target_"])
+            object_name = f"{prefix}".replace(".", "_").lower()
+            objects[object_name] = (target_class, target_params)
+            cfg = object_name
+
+    elif isinstance(cfg, ListConfig):
+        for index, item in enumerate(cfg):
+            item, objects = recursively_walk_and_extract_hydra_targets(item, objects, prefix=f"{prefix}_{index}")
+            cfg[index] = item
+    else:
+        print(f"Skipping {cfg}")
+        pass
+    return cfg, objects
+
+
+def convert_from_recipe(cfg: DictConfig, output_script_path: str):
+    cfg = Trainer._trigger_cfg_modifying_callbacks(cfg)
+    OmegaConf.resolve(cfg)
+
+    device = get_param(cfg, "device")
+    multi_gpu = get_param(cfg, "multi_gpu")
+
+    if multi_gpu is False:
+        multi_gpu = MultiGPUMode.OFF
+    num_gpus = get_param(cfg, "num_gpus")
+    print(device, multi_gpu, num_gpus)
+
+    train_dataloader = get_param(cfg, "train_dataloader")
```

Comment:
This code was taken from `train_from_recipe` method, so we should be fine. In fact, signature of the method looks like this: `def get(name: str = None, dataset_params: Dict = None, dataloader_params: Dict = None, dataset: torch.utils.data.Dataset = None) -> DataLoader:`

Reply:
What I meant is, suppose we have in dataset_params.yaml:
```
train_dataloader:
train_dataloader_params:
  dataset: user_dataset
  shuffle: True
  batch_size: 64
  num_workers: 8
  drop_last: True
  collate_fn: DetectionCollateFN
```

This is fine, because one can just register a dataset, then insde dataloaders.get it would intialize it and pass it to DataLoader's init.

But in line 112 you got:
```
        name="{train_dataloader}",
```
Which will become:
```
name="None"
```

Am I wrong ?

Reply:
Fixed in 5b8089d9a8819c45de5673ca9806833ebca06a8c